pkg/flow/adapter/synchronizer/adapter.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synchronizer
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+)
+
+var _ pkgadapter.Adapter = (*adapter)(nil)
+
+type adapter struct {
+	ceClient cloudevents.Client
+	logger   *zap.SugaredLogger
+
+	correlationKey  *correlationKey
+	responseTimeout time.Duration
+
+	sessions *storage
+	sinkURL  string
+	bridgeID string
+}
+
+// NewAdapter returns adapter implementation.
+func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*envAccessor)
+	logger := logging.FromContext(ctx)
+
```

Comment:
Not asking for this change now, or there to be any hold of any sort. But I would like to comment that, moving forward, It would be nice to see us using the `Replier` package for everything that replies.